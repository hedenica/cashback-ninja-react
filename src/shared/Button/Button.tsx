import { ButtonHTMLAttributes } from 'react'
import clsx from 'clsx'
import styles from './button.module.scss'

interface ButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
  className?: string;
  size: string;
  color: string;
}

export function Button (props: ButtonProps) {
  const { className, children, size, color, ...remainingProps } = props

  const types = {
    size: `--${size}`,
    color: `--${color}`,
  }

  return (
    <button
      className={clsx(
        className,
        styles.button,
        size && [styles[types.size]],
        color && [styles[types.color]],
      )}
      {...remainingProps}
    >
      {children}
    </button>
  )
}
