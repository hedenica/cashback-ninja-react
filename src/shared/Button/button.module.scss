.button {
  display: flex;
  align-items: center;
  justify-content: center;
  border: none;
  font-weight: bold;
  border-radius: 8px;
}

// PROPS

.--yellow {
  background: var(--yellow);
}

.--red {
  background: var(--red);
}

.--small {
  width: 3rem;
  height: 3rem;
  box-shadow: 0px 0.2rem 1.5rem rgb(0 0 0 / 15%);
}

.--large {
  width: 15rem;
  height: 3rem;
  box-shadow: 0px 1rem 1.5rem rgb(0 0 0 / 20%);

  img {
    margin-right: 0.3rem;
  }
}