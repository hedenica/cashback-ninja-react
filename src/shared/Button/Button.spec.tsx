import '@testing-library/jest-dom'
import { render, screen } from '@testing-library/react'

import { Button } from '../Button'

describe('Button', () => {
  it('should render properly', () => {
    const { container } = render(
      <Button size='small' color='yellow'>
        Button Component
      </Button>,
    )

    expect(container).toMatchSnapshot()
  })

  it('should render in yellow', () => {
    render(
      <Button size='small' color='yellow'>
        Button Component
      </Button>,
    )

    expect(screen.getByRole('button')).toHaveClass('--yellow')
  })

  it('should render in smalls size', () => {
    render(
      <Button size='small' color='yellow'>
        Button Component
      </Button>,
    )

    expect(screen.getByRole('button')).toHaveClass('--small')
  })
})
