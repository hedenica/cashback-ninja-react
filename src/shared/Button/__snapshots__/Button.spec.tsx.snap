// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render properly 1`] = `
<div>
  <button
    class="button --small --yellow"
  >
    Button Component
  </button>
</div>
`;
